packages/labs/signals/README.md

```diff
@@ -0,0 +1,208 @@
+   so that only signals that miht have changed and have been read are
+   re-computd. This can help perform minimal computations and DOM updates when
+   doing small updates to large signal graphs or UIs.
+
+5. Signal auto-tracking can reduce the need for component-specific lifecycle
+   APIs. For example, rather than having lifecycle callbacks for when updates
+   have happened, or when specific reactive properties have changed, any code
+   could create a reactive effect that simple accesses the signals it uses, and
+   is automatically re-run when they change.
+
+6. Signals may allow for interoperable _synchronous_ and _batched_ DOM updates.
+   There are ways to respond to signal changes synchronously but also batched,
+   so if reactive properties were backed by signals, an element could re-render
+   itself once a batch of them had been updated. Elements could take care to
+   update children inside of batches, meaning entire subtrees could be updated
+   synchrously. The batching mechanism isn't standard yet, but could be an
+   extension to the proposal.
+
+Signals are a natural fit for Lit. A LitElement render method is already
+somewhat like a computed signal in that it is computed based on updates to
+inputs (reactive properties). The difference between Lit renders and signals is
+that in Lit the data flow is push-based, rather than pull-based as in signals.
+Lit elements react when changes are pushed into them, whereas signals
+automatically subscribe to the other signals they access. But these approaches
+are very compatible, and we can easily make elements subscribe to the signals
+they access and trigger an update with an integration library like this one.
+
+## Usage
+
+### SignalWatcher
+
+`SignalWatcher` is the core of signals integration. It's a mixin that makes an
+element watch all signal accesses during the element's reactive update
+lifecycle, then triggers an element update when signals change. This includes
+signals read in `shouldUpdate()`, `willUpdate()`, `update()`, `render()`,
+`updated()`, `firstUpdated()`, and reactive controller's `hostUpdate()` and
+`hostUpdated()`.
+
+This effectively makes the the return result of `render()` a computed signal.
+
+```ts
+import {LitElement, html} from 'lit';
+import {customElement, property} from 'lit';
+import {SignalWatcher, signal} from '@lit-labs/preact-signals';
+
+const count = signal(0);
+
+@customElement('signal-example')
+export class SignalExample extends SignalWatcher(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${count.get()}</p>
+      <button @click=${this._onClick}>Increment<button></button></button>
+    `;
+  }
+
+  private _onClick() {
+    count.set(count.get() + 1);
+  }
+}
+```
+
+Elements should not _write_ to signals in these lifecycle methods or they might
+cause an infinite loop.
+
+### watch() directive
```

Comment:
I don't want to recommend it because it's not going to work with the type-checking tools until they're updated. They're going to complain that State<string> is not assignable to string and such.